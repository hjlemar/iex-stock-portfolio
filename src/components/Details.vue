<template>
  <div>
    <div v-if='data'>
      <app-delayed-quote-detail :delayedQuote="data"></app-delayed-quote-detail>
      <app-graph :chart="chart"></app-graph>
    </div>
    <div v-else>
      <p>No data found</p>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import DelayedQuoteDetail from './DelayedQuoteDetail';
import Graph from './Graph';

export default {
  components: {
    appDelayedQuoteDetail: DelayedQuoteDetail,
    appGraph: Graph,
  },
  computed: {
    ...mapGetters([
      'date',
      'data',
      'chart',
    ]),
  },
};
</script>

<style>

</style>

