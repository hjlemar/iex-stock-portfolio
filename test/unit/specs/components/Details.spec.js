import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Details from '@/components/Details';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('Details.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        date: '2018-02-01',
      },
      getters: {
        date: jest.fn(),
        data: jest.fn(),
      },
    });
  });

  it('should render component and match snapshot', () => {
    const wrapper = mount(Details, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('date should change to 2018-02-02', () => {
    const wrapper = mount(Details, {
      store,
      localVue,
      computed: {
        date: () => '2018-02-02',
      },
    });

    expect(wrapper.html()).toBe('<div><div><p>No data found</p></div></div>');
  });
});
