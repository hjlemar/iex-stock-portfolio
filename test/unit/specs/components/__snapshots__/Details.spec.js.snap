// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details.vue should render component and match snapshot 1`] = `
<div>
    <div>
        <p>No data found</p>
    </div>
</div>
`;
